dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/LogsFeatureTest.kt

```diff
@@ -419,133 +421,95 @@ internal class LogsFeatureTest {
     }
 
     @Test
-    fun `𝕄 write crash log event 𝕎 onReceive() { with throwable }`(
+    fun `𝕄 write crash log event and not wait 𝕎 onReceive() { JVM crash, timeout }`(
         @StringForgery fakeThreadName: String,
         @LongForgery fakeTimestamp: Long,
         @StringForgery fakeMessage: String,
         @StringForgery fakeLoggerName: String,
         forge: Forge
     ) {
         // Given
+        whenever(mockLogsFeatureScope.withWriteContext(any(), any())) doAnswer {
+            val callback = it.getArgument<(DatadogContext, EventBatchWriter) -> Unit>(1)
+            val executor = Executors.newSingleThreadExecutor()
+            executor.execute {
+                Thread.sleep(LogsFeature.MAX_WRITE_WAIT_TIMEOUT_MS + 200)
+                callback.invoke(fakeDatadogContext, mockEventBatchWriter)
+            }
+        }
         testedFeature.dataWriter = mockDataWriter
         val fakeThrowable = forge.aThrowable()
-        val fakeAttributes = forge.exhaustiveAttributes()
         val event = mapOf(
-            "type" to "crash",
+            "type" to "jvm_crash",
             "threadName" to fakeThreadName,
             "timestamp" to fakeTimestamp,
             "message" to fakeMessage,
             "loggerName" to fakeLoggerName,
-            "attributes" to fakeAttributes,
             "throwable" to fakeThrowable
         )
 
         // When
         testedFeature.onReceive(event)
 
         // Then
-        argumentCaptor<LogEvent> {
-            verify(mockDataWriter).write(eq(mockEventBatchWriter), capture())
+        verify(mockLogsFeatureScope).withWriteContext(any(), any())
```

Comment:
I don't see where are you verifying that you are writing the log here.

Reply:
Nothing will be written because it is the test for the timeout - we check that we can complete the test (caller thread is not blocked forever) and by the time we complete nothing is written yet (`verifyZeroInteractions(mockDataWriter)` checks that), because write operation is still hanging.

Reply:
I think in this case maybe you want to update the test function name ?